.pre-commit-config.yaml

```diff
@@ -5,76 +5,90 @@ repos:
   - repo: https://github.com/pre-commit/pre-commit-hooks
     rev: v4.3.0
     hooks:
-      - id: trailing-whitespace
+      - id: fix-byte-order-marker
+      - id: mixed-line-ending
       - id: end-of-file-fixer
-        exclude: ^.*\.egg-info/
+        exclude: \.egg-info/[^/]+
+      - id: trailing-whitespace
       - id: check-merge-conflict
       - id: check-case-conflict
+      - id: check-symlinks
+      - id: destroyed-symlinks
       - id: check-json
       - id: check-toml
-        exclude: tests/fixtures/invalid_lock/poetry\.lock
+        exclude: fixtures/invalid_lock
       - id: check-yaml
       - id: pretty-format-json
         args: [--autofix, --no-ensure-ascii, --no-sort-keys]
+      - id: check-vcs-permalinks
       - id: check-ast
       - id: debug-statements
       - id: check-docstring-first
+      - id: name-tests-test
+        args: [--pytest-test-first]
+        exclude: (compat|helpers|types|utils)\.py|fixtures/[^/]+
+      - id: no-commit-to-branch
 
   - repo: https://github.com/pre-commit/pygrep-hooks
     rev: v1.9.0
     hooks:
+      - id: python-check-blanket-noqa
+      - id: python-check-blanket-type-ignore
```

Comment:
In the `mypy` config we have enabled `ignore-without-code`. So this check here would redundant.

Reply:
It is strictly speaking so, but I think this is nice as a way to give early feedback (many contributors forget to run mypy locally).

Reply:
I think the argument for removing flake8-annotations is that we have to suppress/special case a decent number of issues it reports; whereas this is free and doesn't require any special casing.